### PR TITLE
Showed logged in user as comment author

### LINKED ISSUE #760: Show correct autenticated user name in comments
Follow-up for #742 as #754 has been introduced.

Make sure that when authentication is turned off, `Anonymous` is put as the comment's _author_.

### PR DESCRIPTION
**Pull request by dkrupp**
Closes #760.

### CODE REVIEW

File: libcodechecker/server/client_db_access_handler.py [hunk 1]
```diff
@@ -781,10 +781,11 @@ def addComment(self, report_id, comment_data):
         try:
             report = session.query(Report).get(report_id)
             if report:
+                user = 'Anonymous'
```
Comment (whisperity) [inline libcodechecker/server/client_db_access_handler.py]: Simplify statement as

```py
user = self.__auth_session.user if self.__auth_session else "Anonymous"
```

File: libcodechecker/server/client_db_access_handler.py [hunk 2]
```diff
@@ -816,9 +817,18 @@ def updateComment(self, comment_id, content):
             Update the given comment message with new content.
         """
         session = self.__session
+        user = 'Anonymous'
```
Comment (whisperity) [inline libcodechecker/server/client_db_access_handler.py]: Statement can be simplified.

File: libcodechecker/server/client_db_access_handler.py [hunk 3]
```diff
@@ -846,10 +856,15 @@ def removeComment(self, comment_id):
         """
             Remove the comment.
         """
+        user = 'Anonymous'
```
Comment (whisperity) [inline libcodechecker/server/client_db_access_handler.py]: Statement can be simplified.

File: tests/libtest/codechecker.py [hunk 4]
```diff
@@ -7,9 +7,9 @@
 import shlex
 import subprocess
 import time
+import json
```
Comment (whisperity) [inline tests/libtest/codechecker.py]: Import order mismatch.

File: tests/functional/comment/__init__.py [hunk 5]
```diff
@@ -92,25 +95,57 @@ def setup_package():
...
+    print("Analyzing test project was successful.")
 
     codechecker_cfg['run_names'] = [test_project_name]
-
     test_config['codechecker_cfg'] = codechecker_cfg
-
     env.export_test_cfg(TEST_WORKSPACE, test_config)
 
 
+def enable_auth(workspace):
```
Comment (whisperity) [inline tests/functional/comment/__init__.py]: Perhaps this method should be made generally available under `libtest` somewhere.

File: libcodechecker/libclient/authentication_helper.py [hunk 6]
```diff
@@ -50,7 +50,10 @@ def wrapper(self, *args, **kwargs):
                 if reqfailure.error_code == shared.ttypes.ErrorCode.DATABASE:
                     print('Database error on server')
                     print(str(reqfailure.message))
-                if reqfailure.error_code == shared.ttypes.ErrorCode.PRIVILEGE:
+                if reqfailure.error_code == shared.ttypes.ErrorCode.AUTH_DENIED:
```
Comment (whisperity) [inline libcodechecker/libclient/authentication_helper.py]: Line too long, makes tests fail.

File: tests/functional/comment/__init__.py [hunk 7]
```diff
@@ -4,11 +4,13 @@
 #   This file is distributed under the University of Illinois Open Source
 #   License. See LICENSE.TXT for details.
 # -----------------------------------------------------------------------------
+from time import sleep
```
Comment (Xazax-hun) [inline tests/functional/comment/__init__.py]: This import might be unused.

File: libcodechecker/server/client_db_access_handler.py [hunk 8]
```diff
@@ -846,10 +856,15 @@ def removeComment(self, comment_id):
         """
             Remove the comment.
```
Comment (csordasmarton) [inline libcodechecker/server/client_db_access_handler.py]: Please replace the documentation of this function to this:
```
Remove the comment. We allow comments to be removed by it's
original author only, except for Anyonymous comments that can be
updated by anybody.
```

File: libcodechecker/server/client_db_access_handler.py [hunk 9]
```diff
@@ -816,9 +817,18 @@ def updateComment(self, comment_id, content):
             Update the given comment message with new content.
```
Comment (csordasmarton) [inline libcodechecker/server/client_db_access_handler.py]: Please replace the documentation of this function to this:
```
Update the given comment message with new content. We allow
comments to be updated by it's original author only, except for
Anyonymous comments that can be updated by anybody.
```

**Comment by csordasmarton** [comment]
The code looks good. Fix the test cases:
```bash
tests/functional/comment_unauth/test_comment_unauth.py:112:53: W291 trailing whitespace
tests/functional/comment_unauth/test_comment_unauth.py:114:1: W293 blank line contains whitespace
tests/functional/comment_unauth/test_comment_unauth.py:114:1: W391 blank line at end of file
```

**Pull request merged**